File: swingLotto/src/main/java/com/eastgrams/swinglotto/model/LottoDrawServiceResult.java
package com.eastgrams.swinglotto.model;

import com.google.gson.annotations.SerializedName;
import java.text.DecimalFormat;

public class LottoDrawServiceResult {
    private static final String SUCCESS_RESPONSE = "success";
    private static final DecimalFormat formatter = new DecimalFormat("#,###");

    @SerializedName("totSellamnt") public long totalSellAmount;
    @SerializedName("returnValue") public String result; // 요청결과
    @SerializedName("drwNoDate") public String drawDate; // 날짜
    @SerializedName("firstWinamnt") public long firstWinAmount; //1등 상금액
    @SerializedName("firstPrzwnerCo") public int firstWinCount; // 1등 당첨인원
    @SerializedName("firstAccumamnt") public long firstWinTotalAmount;
    @SerializedName("drwtNo1") public int drawNo1;// 로또번호 1
    @SerializedName("drwtNo2") public int drawNo2;// 로또번호 2
    @SerializedName("drwtNo3") public int drawNo3;// 로또번호 3
    @SerializedName("drwtNo4") public int drawNo4;// 로또번호 4
    @SerializedName("drwtNo5") public int drawNo5;// 로또번호 5
    @SerializedName("drwtNo6") public int drawNo6;// 로또번호 6
    @SerializedName("bnusNo") public int bonusNo;// 로또 보너스 번호
    @SerializedName("drwNo") public int drawNo;// 로또회차

    public boolean isSuccess() {
        return SUCCESS_RESPONSE.equals(result);
    }

    public String firstAccumamntString() {
        return formatter.format(firstWinTotalAmount);
    }

    public String firstWinAmountString() {
        return formatter.format(firstWinAmount);
    }

    @Override
    public String toString() {
        return "LottoDrawServiceResult{" +
                "totalSellAmount=" + totalSellAmount +
                ", result='" + result + '\'' +
                ", drawDate='" + drawDate + '\'' +
                ", firstWinAmount=" + firstWinAmount +
                ", firstWinCount=" + firstWinCount +
                ", firstWinTotalAmount=" + firstWinTotalAmount +
                ", drawNo1=" + drawNo1 +
                ", drawNo2=" + drawNo2 +
                ", drawNo3=" + drawNo3 +
                ", drawNo4=" + drawNo4 +
                ", drawNo5=" + drawNo5 +
                ", drawNo6=" + drawNo6 +
                ", bonusNo=" + bonusNo +
                ", drawNo=" + drawNo +
                '}';
    }
}

File: android_lottoTV/app/src/main/java/com/eastgrams/lottoResultTvApp/data/LottoDrawServiceResult.kt
package com.eastgrams.lottoResultTvApp.data

import com.google.gson.annotations.SerializedName
import java.text.DecimalFormat

data class LottoDrawServiceResult(
    @field:SerializedName("totSellamnt") val totalSellAmount: Long,
    @field:SerializedName("returnValue") val result: String, // 요청결과
    @field:SerializedName("drwNoDate") val drawDate: String, // 날짜
    @field:SerializedName("firstWinamnt") val firstWinAmount: Long, //1등 상금액
    @field:SerializedName("firstPrzwnerCo") val firstWinCount: Int, // 1등 당첨인원
    @field:SerializedName("firstAccumamnt") val firstWinTotalAmount: Long,
    @field:SerializedName("drwtNo1") val drawNo1: Int,// 로또번호 1
    @field:SerializedName("drwtNo2") val drawNo2: Int,// 로또번호 2
    @field:SerializedName("drwtNo3") val drawNo3: Int,// 로또번호 3
    @field:SerializedName("drwtNo4") val drawNo4: Int,// 로또번호 4
    @field:SerializedName("drwtNo5") val drawNo5: Int,// 로또번호 5
    @field:SerializedName("drwtNo6") val drawNo6: Int,// 로또번호 6
    @field:SerializedName("bnusNo") val bonusNo: Int,// 로또 보너스 번호
    @field:SerializedName("drwNo") val drawNo: Int,// 로또회차
) {
    fun isSuccess(): Boolean {
        return SUCCESS_RESPONSE == result
    }

    fun firstAccumamntString(): String {
        return formatter.format(firstWinTotalAmount)
    }

    fun firstWinAmountString(): String {
        return formatter.format(firstWinAmount)
    }

    companion object {
        const val SUCCESS_RESPONSE = "success"
        private val formatter = DecimalFormat("#,###");
    }
}